pkg/deployment/agency/state.go

```diff
@@ -115,3 +119,70 @@ func (d *StateExists) UnmarshalJSON(bytes []byte) error {
 	*d = bytes != nil
 	return nil
 }
+
+type CollectionIterator func(db, col string, planCollection *StatePlanCollection, shard string, plan ShardServers, current ShardServers)
+
+func (s State) IterateOverCollections(i CollectionIterator) {
+	for db, collections := range s.Plan.Collections {
+		for collection, details := range collections {
+			for shard, shardDetails := range details.Shards {
+				currShard := s.Current.Collections[db][collection][shard]
```

Comment:
Shard can be missing on current - add corner case

Reply:
done